docs/transactions-new.md

```diff
@@ -0,0 +1,749 @@
+        Since clients and different Nodes may be at different versions, we need this field to support backward compatibility
+        between clients and nodes.     
+     
+    - `ver` (integer):
+    
+        Transaction version to be able to evolve content.
+        The content of `data` and `metadata` may depend on the version.       
+ 
+    - `data` (dict):
+
+        Transaction-specific data fields (see next sections for each transaction description).  
+       
+    - `metadata` (dict):
+    
+        Metadata as came from the Request.
+
+        - `from` (base58-encoded string):
+             Identifier (DID) of the transaction submitter (client who sent the transaction) as base58-encoded string
+             for 16 or 32 bit DID value.
+             It may differ from `did` field for some of transaction (for example NYM), where `did` is a 
+             target identifier (for example, a newly created DID identifier).
+             
+             *Example*: `from` is a DID of a Trust Anchor creating a new DID, and `did` is a newly created DID.
+             
+        - `reqId` (integer): 
+            Unique ID number of the request with transaction.
+  
+    - `txnMetadata` (dict):
+    
+        Metadata attached to the transaction.    
+        
+         - `version` (integer):
+            Transaction version to be able to evolve `txnMetadata`.
+            The content of `txnMetadata` may depend on the version.  
+        
+        - `creationTime` (integer as POSIX timestamp): 
+            The time when transaction was written to the Ledger as POSIX timestamp.
+            
+        - `seqNo` (integer):
+            A unique sequence number of the transaction on Ledger
+  
+- `reqSignature` (dict):
+
+    Submitter's signature over request with transaction (`txn` field).
+    
+    - `type` (string enum):
+        
+        - ED25519: ed25519 signature
+        - ED25519_MULTI: ed25519 signature in multisig case.
+    
+    - `values` (list): 
+        
+        - `from` (base58-encoded string):
+        Identifier (DID) of signer as base58-encoded string for 16 or 32 bit DID value.
+        
+        - `value` (base58-encoded string):
+         signature value
+
+Please note that all these metadata fields may be absent for genesis transactions.
+
+## Domain Ledger
+
+#### NYM
+Creates a new NYM record for a specific user, trust anchor, steward or trustee.
+Note that only trustees and stewards can create new trust anchors and trustee can be created only by other trusties (see [roles](https://docs.google.com/spreadsheets/d/1TWXF7NtBjSOaUIBeIH77SyZnawfo91cJ_ns4TR-wsq4/edit#gid=0)).
+
+The transaction can be used for 
+creation of new DIDs, setting and rotation of verification key, setting and changing of roles.
+
+- `id` (string):
+
+    Nym's ID as State Trie key (address or descriptive data). It must be unique within the ledger. 
```

Comment:
Can you give an example of `id`? If `did1` is creating (`did1` is a trust anchor) `did2`, what is `id`? Hash of `did1` or hash of `did2`? Also when do you need to look at a NYM in ledger txn and then go to state trie? And if you need state key, do we need to store the key, cant we store a reference to a mechanism like `1->sha256`, `2->sha3`, etc

Reply:
We should use a general `txnId` instead of `id`. It can be used to uniquely identify a txn in the state. So, it can contain a key in the state trie.